### PR TITLE
Update references to "Cray [Inc.]" w.r.t. HPE's acquisition

### PR DESCRIPTION
**Pull request by bradcray**
This takes a pass over our files, looking for references to "Cray Inc." to update them with
respect to HPE's recent acquisition of Cray when possible.  Specifically it:

* updates copyrights to refer to "Hewlett Packard Enterprise Development LP"
* updates contributors currently employed by HPE to reflect their affiliation
* updates other references to "Cray" (the company) or "Cray Inc." to refer to
  "Hewlett Packard Enterprise" or "HPE", occasionally still referring to Cray for
  historical context when appropriate.

Most of these changes were fairly mechanical other than the ones in the doc/, util/, and
test/ directories which is where I applied the most curation.

Notably, it does not make any attempt to update references to "Cray" when referring to
platforms, compilers, or software, both to reflect the names of systems and software that
are already deployed, and to wait and see what the future holds for those names from
a branding perspective.

Not directly related to this logical change (other than that searching for "cray" caused me to find them), while here I updated:

* a URL that still referred to chapel.cray.com to refer to chapel-lang.org instead
* removed some implementation notes from the spec that didn't seem all that relevant anymore

### CODE REVIEW

**Comment by bradcray** [comment]
@gbtitus: Would you take on the thankless job of reviewing this PR?  (though I will thank you for it!)  Unlike most of my PRs, I think that the best way to tackle this would probably be to look at it commit-by-commit (though I wouldn't literally look through every change in every commit, particularly when the copyrights were involved).  As noted in the PR's message, the doc/ and util/ directories are the ones that probably require the most scrutiny, and you might also want to take the approach of grabbing the branch and doing some grepping against it yourself to see what remains.

**Comment by gbtitus** [comment]
The 👍 on your request means "Yes, I'll review" not "Good to go".  It occurred to me after the fact that I might have created some confusion by doing that!

**Comment by gbtitus** [comment]
"Cray" appears as the affiliation of a CHIP author who has moved to HPE with the core team, in `doc/rst/developer/chips/17.rst` and `22.rst`.  Since among other things CHIPs are supposed to be tracking devices for the proposals, should we attempt to maintain up-to-date affiliations as those change and we know about them?  I'm not proposing here that we be proactive and check all the CHIP authors every year to see if they've moved, but it seems worthwhile to update the ones whose moves have been obvious to us.

**Comment by bradcray** [comment]
> should we attempt to maintain up-to-date affiliations as those change and we know about them?

That's a good question.  I passed over this case due to the fact that the CHIPs hadn't been updated recently, figuring that if/when they were, Michael could update his affiliation.  But I could also see making it HPE proactively or `HPE (formerly Cray)` or `HPE/Cray` or something like that.  @mppf since you kicked off the CHIP idea and are the person in question, what do you suggest?

**Comment by mppf** [comment]
I think it makes sense to leave my affiliation as "Cray" for the moment on these CHIPS, with the view that they are currently "historical" documents. If that changes, we will update them.

**Comment by gbtitus** [comment]
I'm trying to remember, do we care about copyright notices in tests?  If so, `git grep -w Cray` in the branch finds the following, which don't have the correct final year.

```
test/library/packages/HDFS/HDFSiterator.chpl: * Copyright 2004-2019 Cray Inc.
test/library/packages/Itertools/Itertools.chpl: * Copyright 2004-2019 Cray Inc.
test/library/packages/csv/CSV.chpl: * Copyright 2004-2019 Cray Inc.
test/library/standard/DataFrames/DataFrames.chpl: * Copyright 2004-2018 Cray Inc.
test/modules/paths/int/ChapelNumLocales.chpl: * Copyright 2004-2019 Cray Inc.
test/runtime/ferguson/testatomics.test.c: * Copyright 2004-2017 Cray Inc.
test/studies/comd/elegant/arrayOfStructs/util/AccumStencilDist.chpl: * Copyright 2004-2017 Cray Inc.
test/users/Deeksha96/StringAlgorithms.chpl: * Copyright 2004-2018 Cray Inc.
test/users/aroonsharma/MyBlockCyclic.chpl:// Copyright (c) 2004-2013, Cray Inc.  (See LICENSE file for more details)
```

**Comment by gbtitus** [comment]
A few files in the branch have copyright notices that differ slightly in phrasing from all the rest.  These have "(c)" after the word "Copyright":

```
man/chpl.rst
man/chpldoc.rst
```

And this one generates the copyright notice in the spec file that generates the module RPM with just the current year, not a range.  That may be okay though, since clearly the RPM file the spec file helps generate cannot have existed prior to the current year.  As an additional nit, there's a comma following that copyright year.  No other copyright notice has that comma.

```
util/build_configs/cray-internal/generate-rpmspec.bash
```

**Comment by gbtitus** [comment]
I looked at the `doc` and `spec` commits specifically, and at `CONTRIBUTORS.md` and `util` less deeply just to get a sense of the changes.  I also looked all the instances found by these commands in the branch, to make sure that the remaining Cray references were appropriate and the new HP ones were well-formed:

```
git grep -w Cray
git grep -w 'Hewlett\|Packard' | grep -v 'Copyright \(199[0-9]\|20[01][0-9]\|2020\)-\(199[0-9]\|20[01][0-9]\|2020\) Hewlett Packard Enterprise Development LP'
```

(The complicated nature of the 2nd of those was just to try to make sure I caught any typos or variation in the years or the copyright phrasing.)

Modulo the comments I've already made, this looks good to me!

**Comment by bradcray** [comment]
> do we care about copyright notices in tests?

We've traditionally, intentionally had the BTR team not update the copyright dates on tests on an annual basis, primarily because we typically don't want to bother asserting copyright on tests (e.g., I don't want to get into the habit of having our one-line hello world, or a test that a developer submits to have to have a half-page copyright at its top).

In this PR, I bypassed them similarly, though it gave me pause.  I'm almost tempted to remove the copyrights from the tests to avoid this kind of confusion, but was also nervous about doing that from the perspective of seeming as though I was tampering with legal notice.  So I ended up leaving them as-is from the perspective of preserving the way they were contributed to the team.

That said, maybe removing them is the right way to go, considering:

* test/library/* probably has copyrights because they were intended to be contributed to the modules/ directory at some point.  But we can add the copyright at that point rather than now.
* I think one or two of the others are cases where an existing file was cloned and the clone carried the copyrights with it
* the remaining ones seem pretty minimal to me

What are your thoughts?

**Comment by gbtitus** [comment]
Late-breaking news: the `COPYRIGHT` file also has a '(c)' after the word 'Copyright' in its copyright notice, and a comma after the copyright year range.  Sorry I didn't notice this until after I'd posted the previous comment.

File: test/arrays/sparse/sparse.chpl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
   Jan 12, 2016: Michael Noakes
 
-  Cray's internal implementation of sparse arrays currently relies on
+  Out internal implementation of sparse arrays currently relies on
```
Comment (cassella) [inline test/arrays/sparse/sparse.chpl]: *Our

**Comment by gbtitus** [comment]
> What are your thoughts?

I think they probably could be removed, given the 3 considerations you listed.  The only thing that might give me pause is: if someone else grabs one or more of these and slaps their own copyright onto them, do they then "own" them?  As an admittedly borderline silly example, could someone take ra.chpl, add a copyright and restrictive licensing, start selling that program, and tell us we had to remove it from our repo?

**Comment by mppf** [comment]
>  As an admittedly borderline silly example, could someone take ra.chpl, add a copyright and restrictive licensing, start selling that program, and tell us we had to remove it from our repo?

No, because our entire repo has a COPYRIGHT file and Apache license clearly indicated. As far as I understand these things, anyway.

**Comment by gbtitus** [comment]
> No, because our entire repo has a COPYRIGHT file and Apache license clearly indicated.  ...

That would seem to imply we could remove nearly all the other copyright notices too, though.  Perhaps this is an area we should just whistle past, for the purposes of today's PR.   :-)

**Comment by bradcray** [comment]
> >  As an admittedly borderline silly example, could someone take ra.chpl, add a copyright and restrictive licensing, start selling that program, and tell us we had to remove it from our repo?
> 
> No, because our entire repo has a COPYRIGHT file and Apache license clearly indicated. As far as I understand these things, anyway.

I think there is a potential, minor, risk here and that people could choose to add copyrights to specific files if they were worried about it (which could be a reason to keep preserving the ones that are there).  If such a case came up, I think we'd also have the benefit of having a paper trail in the form of GitHub showing that we had and committed the file before they'd asserted their copyright (assuming they didn't create a false paper trail of their own to say otherwise).

**Comment by bradcray** [comment]
@ben-albrecht: Do you have intuition about what to do w.r.t. the copyrights in test/library/packages/Itertools/Itertools.chpl and test/library/packages/csv/CSV.chpl (where the two main approaches we're debating are "remove the copyrights" or "continue to let them go stale over time)?

**Comment by bradcray** [comment]
@ben-albrecht: I ended up making a call after thinking about it some more.  Let me know if you disagree.

File: test/arrays/sparse/sparse.chpl (hunk 1 above)
Comment (bradcray) [reply test/arrays/sparse/sparse.chpl]: Argh!

Comment (bradcray) [reply test/arrays/sparse/sparse.chpl]: (Thanks!)

**Comment by bradcray** [comment]
> A few files in the branch have copyright notices that differ slightly in phrasing from all the rest.

I think we had the man page copyright format mimic what we were seeing in gcc's man page.  But being consistent with our other copyrights is probably the right thing to do.

 > And this one generates the copyright notice in the spec file that generates the module RPM with just the current year, not a range.

I think I'll change it to a range, just for consistency.  By a similar argument, each file should start from the year it was created rather than the 2004 that everything is using, but I don't want to start down that path...

**Comment by bradcray** [comment]
@gbtitus : If you could spot-check the last few commits, overlooking some sloppiness in them, as a final sanity check, I'd appreciate it.  This should do it `git diff 2f8e142..c960c90` though I had a stray commit of a locally-modified execopts file in there that you should ignore (diffing against master, I believe I've got it back to its original state).

**Comment by ben-albrecht** [comment]
> @ben-albrecht: Do you have intuition about what to do w.r.t. the copyrights in test/library/packages/Itertools/Itertools.chpl and test/library/packages/csv/CSV.chpl (where the two main approaches we're debating are "remove the copyrights" or "continue to let them go stale over time)?

I don't think it matters much. We'll update the copyrights whenever we promote to standard/package modules anyway.

**Comment by gbtitus** [comment]
> If you could spot-check the last few commits, ...

They look good and I've re-approved.

**Pull request merged**